### PR TITLE
Fixed bug in MULTI_SZ processing

### LINKED ISSUE #2438: Querying registry keys that are REG_MULTI_SZ causes osquery to crash
Like the title says, querying registry keys that are of type REG_MULTI_SZ causes osquery to crash.

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**
This fixes #2438

### CODE REVIEW

**Comment by PoppySeedPlehzr** [comment]
ok to test

File: osquery/tables/system/windows/registry.cpp [hunk 1]
```diff
@@ -203,6 +203,7 @@ void queryKey(const std::string& hive,
                  _TRUNCATE);
     }
 
+    BYTE* multiSzStrPtr = bpDataBuff;
```
Comment (theopolis) [inline osquery/tables/system/windows/registry.cpp]: This variable name is weird.

File: osquery/tables/system/windows/registry.cpp [hunk 2]
```diff
@@ -231,9 +232,9 @@ void queryKey(const std::string& hive,
       r["data"] = std::string(regLinkStr);
       break;
     case REG_MULTI_SZ:
-      while (cnt <= cbMaxValueData) {
-        std::string s((char*)bpDataBuff);
-        bpDataBuff += s.size();
+      while (*multiSzStrPtr != 0x00) {
```
Comment (theopolis) [inline osquery/tables/system/windows/registry.cpp]: Set a `\x00` explicitly after this container is filled in.

**Comment by facebook-github-bot-6** [comment]
@PoppySeedPlehzr updated the pull request - [view changes](https://github.com/facebook/osquery/pull/2439/files/e4ae2189457c46ecc726813ea3760d2203bba3ab..6b25e0495695d389a5851a8b5e0929eb0885cd5d)

**Comment by theopolis** [comment]
ok to test

**Comment by osqueryer** [comment]
:-1: The commit 6b25e0495695d389a5851a8b5e0929eb0885cd5d (Job results: [3197](https://jenkins.osquery.io/job/osqueryPullRequestBuild/3197/)) failed one or more tests (**OS X/Linux**).

**Comment by theopolis** [comment]
Build error is unrelated to OSes, there was an internal Jenkins error that caused a connection drop to the build host.

```
-- Build files have been written to: /home/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/centos7/build/centos7
Running tests...
Test project /home/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/centos7/build/centos7
    Start 1: osquery_tests
1/9 Test #1: osquery_tests ....................   Passed    0.15 sec
    Start 2: osquery_additional_tests
Agent went offline during the build
ERROR: Connection was broken: java.io.IOException: Unexpected termination of the channel
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:73)
Caused by: java.io.EOFException
	at java.io.ObjectInputStream$PeekInputStream.readFully(ObjectInputStream.java:2351)
	at java.io.ObjectInputStream$BlockDataInputStream.readShort(ObjectInputStream.java:2820)
	at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:804)
	at java.io.ObjectInputStream.<init>(ObjectInputStream.java:302)
	at hudson.remoting.ObjectInputStreamEx.<init>(ObjectInputStreamEx.java:48)
	at hudson.remoting.AbstractSynchronousByteArrayCommandTransport.read(AbstractSynchronousByteArrayCommandTransport.java:34)
	at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:59)

Build step 'Execute shell' marked build as failure
```

**Pull request merged**